src/xcode/ENA/ENA/Source/Services/HealthCertificateValidation/HealthCertificateValidationService.swift

```diff
@@ -152,10 +145,13 @@ final class HealthCertificateValidationService: HealthCertificateValidationProvi
 							completion(.failure(.VALUE_SET_CLIENT_ERROR))
 						}
 					}
-					
 				}, receiveValue: { [weak self] valueSets in
+					guard let self = self else { return }
+
 					Log.info("Successfully received value sets. Proceed with downloading acceptance rules...", log: .vaccination)
-					self?.downloadAcceptanceRules(
+					self.rulesDownloadService = RulesDownloadService(store: self.store, client: self.client)
```

Comment:
here again, the rulesDownloadService should not be initialized everytime we value sets changes.